runner/lib/runAssignmentInSecretAgent.ts

```diff
@@ -1,68 +1,111 @@
 import { Agent } from 'secret-agent';
+import Core from '@secret-agent/core';
+import { IRunnerFactory, IRunner } from '../interfaces/runner';
 import IAssignment from '@double-agent/collect-controller/interfaces/IAssignment';
 import ISessionPage from '@double-agent/collect/interfaces/ISessionPage';
 
-let lastPage: ISessionPage;
+class SecretAgentRunnerFactory implements IRunnerFactory {
+  connectionServerPort: number;
+  connectionToCore: { host: string };
 
-export default async function runAssignmentInSecretAgent(agent: Agent, assignment: IAssignment) {
```

Comment:
The convention we've used in the codebase is that a file uses the name of the class or function that is the default "thing" in the file. We've stuck with default exports for most classes even though they are somewhat undesirable in module exports vs require exports.

Reply:
eg, this file should be renamed to SecretAgentRunnerFactory and the default export should be `SecretAgentRunnerFactory`

Reply:
We've been generally keeping to a single export when it's a specific class or interface. Occasionally we will have inline interfaces and classes that don't get exported. We do sometimes have a related interface get exported as a named export alongside a default export that the class is named after.

If you have a utility file with several functions, we've been using the standard of that being a lower cased filename.